prometheus/src/cluster_metrics.rs

```diff
@@ -1,12 +1,85 @@
 use solana_gossip::cluster_info::ClusterInfo;
 use solana_runtime::bank::Bank;
+use solana_sdk::{clock::Slot, pubkey::Pubkey};
+use solana_vote_program::vote_state::VoteState;
 
 use crate::{
     utils::{write_metric, Metric, MetricFamily},
     Lamports,
 };
 use std::{io, sync::Arc};
 
+struct ValidatorVoteInfo {
+    vote_address: Pubkey,
+    balance: Lamports,
+    last_vote: Slot,
+}
+
+impl ValidatorVoteInfo {
+    fn new_from_bank(bank: &Arc<Bank>, identity_pubkey: &Pubkey) -> Option<Self> {
+        let vote_accounts = bank.vote_accounts();
+        let vote_state_default = VoteState::default();
+        vote_accounts
+            .iter()
+            .filter_map(|(&vote_pubkey, (_activated_stake, account))| {
+                let vote_state = account.vote_state();
+                let vote_state = vote_state.as_ref().unwrap_or(&vote_state_default);
+                if identity_pubkey != &vote_state.node_pubkey {
+                    return None;
+                }
+                let last_vote = if let Some(vote) = vote_state.votes.iter().last() {
+                    Some(vote.slot)
+                } else {
+                    None
+                }?;
+                let vote_balance = Lamports(bank.get_balance(&vote_pubkey));
+                Some(ValidatorVoteInfo {
+                    vote_address: vote_pubkey,
+                    balance: vote_balance,
+                    last_vote,
+                })
+            })
+            .next()
+    }
+
+    fn write_prometheus<W: io::Write>(&self, out: &mut W) -> io::Result<()> {
+        write_metric(
+            out,
+            &MetricFamily {
+                name: "solana_node_vote_public_key_info",
+                help: "The current Solana node's vote public key",
+                type_: "count",
+                metrics: vec![Metric::new(1).with_label("vote", self.vote_address.to_string())],
+            },
+        )?;
+        // We can use this metric to track if the validator is making progress
+        // by voting on the last slots.
+        write_metric(
+            out,
+            &MetricFamily {
+                name: "solana_node_last_vote_slot",
+                help:
+                    "The voted-on slot of the validator's last vote that got included in the chain",
+                type_: "gauge",
+                metrics: vec![
+                    Metric::new(self.last_vote).with_label("pubkey", self.vote_address.to_string())
```

Comment:
```suggestion
                    Metric::new(self.last_vote).with_label("vote_account", self.vote_address.to_string())
```

Ideally we would add the associated identity account as well. With the Lido for Solana alerts, in some cases I wanted to build a message based on the identity account, to be able to link to a block explorer, but we didn’t have that information available in the metrics.